File: trunk/net/_template/printform.vb
Namespace frm._template
    Public Class printform
        'Inherits saiiform
        Inherits repsol.forms.template.informe.ReportForm

        Protected WithEvents btRefresh As System.Windows.Forms.Button
        'Protected WithEvents SplitContainer1 As System.Windows.Forms.SplitContainer
        'Public WithEvents viewer As Microsoft.Reporting.WinForms.ReportViewer

        Private _hasParameters As Boolean

        Public Sub New()
            MyBase.New()
            InitializeComponent()

            Me.RememberUserPreferences = True
        End Sub
        Public Sub New(ByVal hasParameters As Boolean)
            MyBase.New(hasParameters)
            InitializeComponent()

            Me.RememberUserPreferences = True
            Me.tsbVerParametros.Enabled = True

        End Sub

        Private Sub InitializeComponent()
            Me.btRefresh = New System.Windows.Forms.Button
            Me.SplitContainer1.Panel1.SuspendLayout()
            Me.SplitContainer1.Panel2.SuspendLayout()
            Me.SplitContainer1.SuspendLayout()
            Me.SuspendLayout()
            '
            'SplitContainer1
            '
            Me.SplitContainer1.Location = New System.Drawing.Point(0, 25)
            '
            'SplitContainer1.Panel1
            '
            Me.SplitContainer1.Panel1.Controls.Add(Me.btRefresh)
            Me.SplitContainer1.Size = New System.Drawing.Size(792, 548)
            '
            'viewer
            '
            Me.viewer.ShowPrintButton = False
            Me.viewer.Size = New System.Drawing.Size(792, 548)
            '
            'btRefresh
            '
            Me.btRefresh.Anchor = CType((System.Windows.Forms.AnchorStyles.Top Or System.Windows.Forms.AnchorStyles.Right), System.Windows.Forms.AnchorStyles)
            Me.btRefresh.Image = Global.GesTDC.PC.GesTDC.My.Resources.Resources.ico_report
            Me.btRefresh.Location = New System.Drawing.Point(721, 39)
            Me.btRefresh.Name = "btRefresh"
            Me.btRefresh.Size = New System.Drawing.Size(65, 55)
            Me.btRefresh.TabIndex = 0
            Me.btRefresh.Text = "Actualizar"
            Me.btRefresh.TextImageRelation = System.Windows.Forms.TextImageRelation.ImageAboveText
            Me.btRefresh.UseVisualStyleBackColor = True
            Me.btRefresh.Visible = False
            '
            'printform
            '
            Me.ClientSize = New System.Drawing.Size(792, 573)
            Me.Name = "printform"
            Me.SplitContainer1.Panel1.ResumeLayout(False)
            Me.SplitContainer1.Panel2.ResumeLayout(False)
            Me.SplitContainer1.ResumeLayout(False)
            Me.ResumeLayout(False)
            Me.PerformLayout()

        End Sub

        Private Sub btRefresh_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btRefresh.Click
            Cursor = Cursors.WaitCursor
            VerInforme()
            Cursor = Cursors.Default
        End Sub

        Protected Sub guardaCAError(ByVal ex As Exception, ByVal frm As repsol.forms.template.BaseForm, Optional ByVal mostrarMensaje As Boolean = True)
            Dim caerror As New VO.caerror.caerrorVO
            caerror.CAEDesError = ex.Message
            caerror.CAEFormulario = frm.Name
            caerror.CAEProcedimiento = ex.TargetSite.Name
            caerror.CAEIdCASesion = _common.objs.CASesion.Sesion.IdCASesion
            Dim accAdd As New BL.caerror.doAdd(caerror)
            accAdd.execute()
            If (mostrarMensaje) Then
                repsol.util.messages.ShowError(ex.Message, frm.Text)
            End If
        End Sub

#Region "AppSetting"

        'Protected Overrides Function getUserPreferences() As repsol.util.setting.userpreferences
        '    Dim res As New _common.settings.AppPreferences(MyBase.getUserPreferences())

        '    Return res
        'End Function

#End Region

    End Class
End Namespace

File: trunk/net/gesClinica/gesClinica.app.pc/_template/frm/impresion/PrintForm.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace gesClinica.app.pc.template.frm.impresion
{
    public class PrintForm : repsol.forms.template.informe.ReportForm
    {
        public PrintForm()
            : base()
        {
            InitializeComponent();
        }
        public PrintForm(bool hasParameters)
            : base(hasParameters)
        {
            InitializeComponent();
        }
        private void InitializeComponent()
        {
            System.ComponentModel.ComponentResourceManager resources = new System.ComponentModel.ComponentResourceManager(typeof(PrintForm));
            this.SplitContainer1.Panel2.SuspendLayout();
            this.SplitContainer1.SuspendLayout();
            this.SuspendLayout();
            // 
            // SplitContainer1
            // 
            this.SplitContainer1.Location = new System.Drawing.Point(0, 25);
            this.SplitContainer1.Size = new System.Drawing.Size(792, 548);
            // 
            // viewer
            // 
            this.viewer.ShowPrintButton = false;
            this.viewer.Size = new System.Drawing.Size(792, 548);
            // 
            // PrintForm
            // 
            this.ClientSize = new System.Drawing.Size(792, 573);
            this.Icon = ((System.Drawing.Icon)(resources.GetObject("$this.Icon")));
            this.Name = "PrintForm";
            this.Load += new System.EventHandler(this.PrintForm_Load);
            this.SplitContainer1.Panel2.ResumeLayout(false);
            this.SplitContainer1.ResumeLayout(false);
            this.ResumeLayout(false);
            this.PerformLayout();

        }

        private void PrintForm_Load(object sender, EventArgs e)
        {
            try
            {
                //this.OwnerPrint = true;
                this.OwnerPrint = pc._common.cache.IsOwnerPrint;
            }
            catch (Exception ex)
            {
                _common.messages.ShowError(ex);
            }
        }
    }
}
